tests/v1/tpu/test_topk_topp_sampler.py

```diff
@@ -0,0 +1,120 @@
+# SPDX-License-Identifier: Apache-2.0
+import math
+
+import torch
+
+from vllm.platforms import current_platform
+from vllm.v1.sample.ops.topk_topp_sampler import apply_top_k_top_p_tpu
+
+if current_platform.is_tpu():
+    import torch_xla.core.xla_model as xm
+
+DEVICE = xm.xla_device() if current_platform.is_tpu() else torch.device("cuda")
```

Comment:
I think if the test is under `v1/tpu` we shouldn't test for CUDA but skip if platform is not tpu. Otherwise we can move the test into the shared directory.

Reply:
Yep makes sense. Updated to TPU only.